backend/assignments/2/src/main/java/com/caching/exception/CustomException.java

```diff
@@ -0,0 +1,30 @@
+package com.caching.exception;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * CustomException is a specialized RuntimeException that represents
+ * exceptions specific to the Geocoding and Decoding application.
+ * This class extends RuntimeException and is used to throw custom exceptions
+ * throughout the application with meaningful messages.
+ */
+public class CustomException extends RuntimeException {
```

Comment:
Name the class with a valid exception name. Not just `custom`
Pinpoint the actual exception you intend to catch and present it accordingly